frontend/hooks/useAPI.js

```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+import { API_URL } from "@env";
+
+const useAPI = (endpoint, onSuccess, onError) => {
+    const [loading, setLoading] = useState(false);
+
+    const fetchData = async () => {
+        try {
+            setLoading(true);
+            const response = await fetch(`${API_URL}/${endpoint}`);
```

Comment:
actual fetching is performing here
we set loading as true
make fetching
run onSuccess callback
set loading as false

if error - run onError callback